src/compiler/crystal/semantic/ast.cr

```diff
@@ -31,6 +31,35 @@ module Crystal
         false
       end
     end
+
+    def supports_autocast?
+      case self
+      when NumberLiteral
+        true
+      when SymbolLiteral
+        true
+      else
+        case self.type?
+        when IntegerType
+          true
+        else
+          false
+        end
```

Comment:
```suggestion
      when NumberLiteral then true
      when SymbolLiteral then true
      else
        self.type?.is_a?(IntegerType)
```

Reply:
Thanks, I used a different style. I'm still not convinced it's more readable than the original way which let you consider each case separately, and also add more cases later on, but meh.